igor-web/src/main/groovy/com/netflix/spinnaker/igor/travis/TravisBuildMonitor.groovy

```diff
@@ -92,6 +100,19 @@ class TravisBuildMonitor implements PollingMonitor{
                 }
             } as Action0, 0, pollInterval, TimeUnit.SECONDS
         )
+
+        worker.schedulePeriodically(
+            {
+                if (isInService()) {
```

Comment:
another way to ensure atomicity is to use lua scripts and let redis deal with it